### PR TITLE
Feature/refactor to class

### PR DESCRIPTION
**Pull request by MikeGertrudes**
- refactor to es6 class
- add es6 linting configuration for eslint

### CODE REVIEW

File: src/omni-slider.js [hunk 1]
```diff
@@ -1,532 +1,540 @@
-/* Constructor function
- * elementContainer - this acts as a wrapper for the slider, all of the sliders DOM elements will be transcluded inside the <element> provided
- * options - contains the options for the slider
- */
-function Slider(elementContainer, options) {
+(function() {
```
Comment (MikeGertrudes) [inline src/omni-slider.js]: This would be necessary on it's own, but Babel is wrapping the entire piece in an IIFE with `use strict` on build as it is.

**Comment by MikeGertrudes** [comment]
With the addition of `Class` and the necessary indentation, this diff looks a lot heavier than it is. It is really just explicitly defining the `constructor` and transforming the `prototype` methods to straight methods on the `Class`. The only strange one is the use of `get` for `defaultOptions`.

File: src/omni-slider.js (hunk 1 above)
Comment (bcVamp) [inline src/omni-slider.js]: Minor question: Do you really need IIFE and `'use strict';` ?

I was able to comment out the outer wrapper of IIFE & strict mode code, could still able to run `npm start` to edit changes and build the correct artifacts, as the highlighted IDE files in the 1st photo.

Then, I was able to verify that the Demo page in the browser then the build artifacts without IIFE & strict mode could run correctly, without any run-time console error, as in 2nd photo.

screen shot 2016-06-05 at 8 21 11 am

screen shot 2016-06-05 at 8 17 03 am

File: package.json [hunk 2]
```diff
@@ -17,6 +17,11 @@
       "es2015"
     ]
   },
+  "eslintConfig": {
```
Comment (bcVamp) [inline package.json]: @MikeGertrudes , nice. I was trying to figure out the eslint config in my local, but I ended up with a nested, ugly looking config object for this `package.json`. Can we consider upgrading to `es2016`?  (Would like to experiment Decorator when this code base supports `es2016`)

File: src/omni-slider.js (hunk 1 above)
Comment (bcVamp) [inline src/omni-slider.js]: Minor question: Do you really need IIFE and `'use strict';` ?

I was able to comment out the outer wrapper of IIFE & strict mode code, could still able to run `npm start` to edit changes and build the correct artifacts, as the highlighted IDE files in the 1st photo.

Then, I was able to verify that the Demo page in the browser then the build artifacts without IIFE & strict mode could run correctly, without any run-time console error, as in 2nd photo.

screen shot 2016-06-05 at 8 21 11 am

screen shot 2016-06-05 at 8 17 03 am

Comment (MikeGertrudes) [inline src/omni-slider.js]: open to not having a `use strict` or an `IIFE` at all. Babel is doing this for it anyway. @chrisrng?

Comment (chrisrng) [inline src/omni-slider.js]: Yes babel is doing it anyways

Comment (thehindenburg) [inline src/omni-slider.js]: this is actually being done by a gulp process, `gulp.iife()`, if you remove that from the build process this shoulw be fine.

Comment (bcVamp) [inline src/omni-slider.js]: Is it possible to keep the gulp took support IIFE, then the code base would get the benefit of following es2015 pattern natural without being wrapped in IIFE?

**Comment by MikeGertrudes** [comment]
@bcVamp @chrisrng a lot of chatter here, what's the action item here?

**Comment by chrisrng** [comment]
@MikeGertrudes adding in a callback function optional param

**Comment by chrisrng** [comment]
@MikeGertrudes believe just removing extra iife and extra use strict

**Comment by MikeGertrudes** [comment]
Ok @chrisrng RESOLVED. Removed `use strict` and `IIFE`. Anything else?

File: src/omni-slider.js [hunk 3]
```diff
@@ -2,528 +2,531 @@
...
+    this.topics = {
+      start: [],
+      moving: [],
+      stop: []
+    };
+
+    // Contains the DOM elements for the slider
+    this.UI = {
+      slider: null,
+      hanldeLeft: null,
```
Comment (chrisrng) [inline src/omni-slider.js]: Merge conflict @mcjffld fixed this typo

**Comment by chrisrng** [comment]
We're good to go except for that one last thing

**Comment by MikeGertrudes** [comment]
@chrisrng should be resolved in this branch, but still - looking to maintain a merge request and not push to `master`. Can you advise?

**Comment by chrisrng** [comment]
Hmm weird.. let me try to merge it anyway

**Comment by chrisrng** [comment]
Fixed.. had to merge in master

**Pull request merged**